tracer/src/Datadog.Trace/Iast/Range.cs

```diff
@@ -61,6 +62,16 @@ public bool IsMarked(SecureMarks marks)
         return (SecureMarks & marks) != NotMarked;
     }
 
+    public bool IsSafeSource(SourceType[]? safeSources)
+    {
+        if (safeSources is null || Source is null)
+        {
+            return false;
+        }
+
+        return safeSources.Contains(Source.Origin);
```

Comment:
nit: this uses linq and IIRC allocates(?) I would prob just unroll
```suggestion
        foreach (var source in safeSources)
        {
            if (Source.Origin == source)
            {
                return true;
            }
        }

        return false
```